src/Address.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace DevMob\Postcodes;
+
+use DevMob\Postcodes\Cache\NullCache;
+use DevMob\Postcodes\Providers\ProviderInterface;
+use Psr\SimpleCache\CacheInterface;
+
+class Address implements ProviderInterface
```

Comment:
You should name this core wrapper the same as the package: `Postcodes`.
Usage will be:

```php
$provider = new PostcodeApiNu(['key' => '...']);
$postcodes = new Postcodes($provider);
$results = $postcodes->lookup('3011ED', '50');
```